addons/mail/static/src/messaging/component/notification_group/notification_group.js

```diff
@@ -0,0 +1,125 @@
+odoo.define('mail.messaging.component.NotificationGroup', function (require) {
+'use strict';
+
+const useStore = require('mail.messaging.component_hook.useStore');
+
+const { Component } = owl;
+
+class NotificationGroup extends Component {
+
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        useStore(props => {
+            const group = this.env.entities.NotificationGroup.get(props.notificationGroupLocalId);
+            return {
+                group,
+            };
+        });
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @returns {mail.messaging.entity.Notification}
+     */
+    get group() {
+        return this.env.entities.NotificationGroup.get(this.props.notificationGroupLocalId);
+    }
+
+    /**
+     * @returns {string|undefined}
+     */
+    image() {
+        if (this.group.notification_type === 'email') {
+            return '/mail/static/src/img/smiley/mailfailure.jpg';
+        }
+    }
```

Comment:
I think it can go either way... This is related to the entity, but it is also a purely visual thing used only in the context of this component.

The only reason why it is in JS in the first place, is because we cannot easily patch XML attribute based on a condition like this. If it was, then it would be in the template, so I think it makes sense to keep it in the component JS?